### PR TITLE
`test_anonymous_read_only`: Use s3_get_object with retry 

### PR DESCRIPTION
**Pull request by sagihirshfeld**
Fixes: #9013

Requires a backport to 4.14

### CODE REVIEW

File: tests/manage/mcg/test_bucket_policy.py [hunk 1]
```diff
@@ -985,62 +986,3 @@ def test_multipart_with_policy(self, mcg_obj, bucket_factory):
         # Completing the Multipart Upload
         logger.info(f"Completing the MP Upload with on bucket: {bucket}")
         complete_multipart_upload(obc_obj, bucket, object_key, upload_id, uploaded_part)
-
```
Comment (udaysk23) [inline tests/manage/mcg/test_bucket_policy.py:988]: @sagihirshfeld I think you mistakenly removed this test definition. Could you please add it back

File: tests/manage/mcg/test_bucket_policy.py [hunk 2]
```diff
@@ -985,62 +986,3 @@ def test_multipart_with_policy(self, mcg_obj, bucket_factory):
...
-            for action in bucket_policies
-            if action.startswith("s3:")
-        ]
-        actions_list = [action.lower() for action in actions_list]
-        missing_policies = [
-            action for action in actions_list if action not in bucket_policies
-        ]
-        assert (
-            not missing_policies
-        ), f"Some bucket_policies are not created : {missing_policies}"
```
Comment (udaysk23) [inline tests/manage/mcg/test_bucket_policy.py:1046]: @sagihirshfeld I think you mistakenly removed this test definition. Could you please add it back

Comment (sagihirshfeld) [reply tests/manage/mcg/test_bucket_policy.py]: True, that was definitely not my intention. Thanks for catching this!

Comment (sagihirshfeld) [reply tests/manage/mcg/test_bucket_policy.py]: Fixed

**Review by udaysk23 — APPROVED** [review]

**Review by keesturam — APPROVED** [review]

**Review by keesturam — APPROVED** [review]

**Review by udaysk23 — APPROVED** [review]

**Pull request merged**